Automattic/jetpack
# My Jetpack: Add Backup 'needs-attention' red bubble and notice banner

This PR add a red bubble and notice/banner when the backup 'needs-attention' status is triggered (when a backup failure is detected).




Screenshot: 

<img width="1306" alt="Screen Shot 2024-12-11 at 08 20 58" src="https://github.com/user-attachments/assets/c32772bb-973b-4639-9ee1-7884b94f834c">


Fixes: https://github.com/Automattic/jetpack-roadmap/issues/1430

## Proposed changes:

* Adds a red bubble when a BAckup failure is detected.
* Displays a red bubble notice/banner (Top of My Jetpack page) with message, primary CTA the says "Read troubleshooting guide" and secondary CTA that says "Contact support".

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-bJN-p2
Design post: p1HpG7-umC-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






**Testing notes:** To test this PR, basically we need to force a backup to fail. I tried various ways to make a backup fail, but most my attempts were unsuccessful. The only way I've been able to make a failed backup is by making the site unaccessible, and then queueing the backup to run from the MC Rewind Debugger. To make the site unaccessible, I simply killed my Jurassic Tube tunneling, but there are other various ways to do this such as by stopping Apache, blocking port 80, inserting `die()` or `exit()` command in index.php, etc..
There is another promising way of making a backup fail due to a "No such file" error, but I believe the method only works on a self-hosted site that is Not running within a Docker container. (I was not able to get it to work on my local dev environment or with Jurassic Ninja (confirmed it does not work on a JN site here: p1733501055164569/1733349201.389539-slack-CS8UYNPEE )
Instructions on how to carry out this method are here: _Generating “No such file” errors during backup_: pbuNQi-1Wi-p2
If you know of any other ways to trigger a backup failure, please let me know!

### Testing steps:

* Run this PR branch locally in a [local Jetpack dev environment](https://github.com/Automattic/jetpack/blob/trunk/docs/development-environment.md#docker-supported-recommended) (via Docker & JurassicTube or Ngrok tunneling).
* Once Jetpack is up and running, ensure it is fully connected. (Also ensure you are NOT sandboxed, and NOT using STORE_SANDOX or anything like that.)
* Purchase a Backup plan (using A11N credits).
* Wait for your first full backup to complete.
* Kill your Jurassic Tube or Ngrok tunneling, so the site is no longer accessible via the tunneling domain.
* Open your site in the MC Rewind Debugger: PCYsg-lNo-p2
* Queue up a new full backup on your site.
* Wait until the full backup completes:
    - in the MC Rewind debugger page you can find the current backup in the "Recent Full Backup Attempts" section and wait for it to show that is done (it should finish with the status 'not-accessible').
    - You can also open the WPCOM API Developer Console: https://developer.wordpress.com/docs/api/console/. (sign in)
        - Check the rewind backups endpoint at `wpcom/v2`, endpoint: `/sites/239657125/rewind/backups?force=wpcom`, and verify the status of the last backup is `not-available`.
* Now re-connect your Jurassic Tube or Ngrok tunneling.
* Open your site ad go to My Jetpack page.
* At the top of My Jetpack you should see a Red bubble and notice banner: (See screenshot)

![Screen Shot 2024-12-18 at 12 17 54](https://github.com/user-attachments/assets/11e14fad-1f0a-4c3c-8885-17eff80ab225)


* Confirm the button to "Read troubleshooting guide" points to: https://jetpack.com/support/backup/troubleshooting-jetpack-backup/
* Confirm the "Contact support" button points to: https://jetpack.com/support/
* Ensure the Notice/banner looks good in all viewport sizes & mobile sizes.
* Look at the Backup product card. The status of the backup card should be "Needs attention". The card should have a red border around it, and the CTA should say, "Troubleshoot" which points to the Jetpack.com Troubleshooting backup page.
* Also the Backup card should say, "The last backup attempt failed." with a tooltip. And also say "Check out our troubleshooting guide" (see screenshot:)
* Click to open the tooltip. Verify it looks like the screenshots below:

![Screen Shot on 2024-12-18 at 12-15-52](https://github.com/user-attachments/assets/5c127a65-bb78-437b-9544-df13d93b0446)

![Screen Shot on 2024-12-18 at 12-16-12](https://github.com/user-attachments/assets/36aea825-ed26-491e-8fcb-89244dd6edbc)

* Next, disconnect Jetpack. The red bubble and notice should now be the "Missing site connection" notice. Verify there are no console errors.
* Now establish a site-only connection.  In this case there should now be 2 red bubbles. 1 for missing user connection, and 1 for Backup needs-attention, but the missing user connection notice is higher priority so it should be showing. Ensure there are no console errors.
* Now re-establish user connection, and you should now see the Backup needs-attention notice again.

Note: In a followup PR I think I should implement allowing the user to optionally close(x) each red bubble notice (persistent for a temporary amount of time).
